staging_vespalib/src/vespa/vespalib/util/singleexecutor.cpp

```diff
@@ -41,12 +41,38 @@ SingleExecutor::addTask(Task::UP task) {
     return wp;
 }
 
+void
+SingleExecutor::wakeupConsumer() {
+    MonitorGuard guard(_consumerMonitor);
```

Comment:
You can skip taking the lock here; consider using std::mutex and std::condition_variable instead.

Reply:
OK, will do, have not used them directly before.

Reply:
[cppreference](https://en.cppreference.com/w/cpp/thread/condition_variable/notify_one) doc on `std::condition_variable` is pretty handy on this subject:

> The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s); in fact doing so is a pessimization, since the notified thread would immediately block again, waiting for the notifying thread to release the lock. However, some implementations (in particular many implementations of pthreads) recognize this situation and avoid this "hurry up and wait" scenario by transferring the waiting thread from the condition variable's queue directly to the queue of the mutex within the notify call, without waking it up.

Reply:
Done